pkg/wallet/account/account.go

```diff
@@ -0,0 +1,265 @@
+package account
+
+import (
+	"crypto/ed25519"
+	"crypto/rand"
+	"fmt"
+
+	"github.com/awnumar/memguard"
+	"github.com/btcsuite/btcutil/base58"
+	"github.com/massalabs/station-massa-wallet/pkg/crypto"
+	"github.com/massalabs/station-massa-wallet/pkg/types"
+	"github.com/massalabs/station-massa-wallet/pkg/types/object"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	AccountLastVersion = 1
+)
+
+type Account struct {
+	Version      *uint8                    `yaml:"Version"`
+	Nickname     string                    `yaml:"Nickname,omitempty"`
+	Address      types.Address             `yaml:"Address,omitempty"`
+	Salt         [16]byte                  `yaml:"Salt,flow"`
+	Nonce        [12]byte                  `yaml:"Nonce,flow"`
+	CipheredData types.EncryptedPrivateKey `yaml:"CipheredData,flow"`
+	PublicKey    types.PublicKey           `yaml:"PublicKey,flow"`
+}
+
+func New(version *uint8, nickname string, address types.Address, salt [16]byte, nonce [12]byte, cipheredData types.EncryptedPrivateKey, publicKey types.PublicKey) (*Account, error) {
+	if version == nil {
+		version = new(uint8)
+		*version = AccountLastVersion
+	}
+
+	if *version > AccountLastVersion {
+		return nil, fmt.Errorf("unsupported account version: %d", *version)
```

Comment:
you should wrap an invalidArgument sentinel error.

Reply:
what about `object.ErrUnsupportedVersion`?

Reply:
I was more thinking of:
```golang
var ErrInvalidParameter = errors.New("invalid parameter")
...
if version > AccountLastVersion || version == UnknownVersion {
   return nil, return fmt.Errorf("%w: invalid version", ErrInvalidParameter)
}
```
This way, on the caller side you can easily detect that an error is about an invalid parameter, to return an http error code 422 for instance, by doing:
```golang
err := account.New(...)
if errors.Is(err, ErrInvalidParameter) {
    // handle the invalid parameter error
}
```